src/commands/autocomplete/script.js

```diff
@@ -41,21 +42,21 @@ compinit;`)
   static async generateAutocompleteCommands ({config, out}: {config: Config, out: Output}) {
     const flatten = require('lodash.flatten')
     try {
-      // TODO: move from cli to client dir if not already present
-      // if (!fs.pathExistsSync(path.join(this.config.dataDir, 'client', 'autocomplete', 'bash', 'heroku'))) {
-      //   const cli = path.join(this.config.dataDir, 'cli', 'autocomplete')
-      //   const client = path.join(this.config.dataDir, 'client', 'autocomplete')
-      //   fs.copySync(cli, client)
-      // }
       const plugins = await new Plugins(out).list()
-      const cmds = plugins.map(p => p.commands.filter(c => !c.hidden).map(c => {
-        let publicFlags = Object.keys(c.flags).filter(flag => !c.flags[flag].hidden).map(flag => `--${flag}`).join(' ')
-        let flags = publicFlags.length ? ` ${publicFlags}` : ''
-        let namespace = p.namespace ? `${p.namespace}:` : ''
-        return `${namespace}${c.id}${flags}`
+      const cmds = await Promise.all(plugins.map(async (p) => {
+        const hydrated = await p.pluginPath.require()
+        const cmds = hydrated.commands || []
+        return cmds.filter(c => !c.hidden).map(c => {
+          const Command = typeof c === 'function' ? c : convertFromV5((c: any))
+          const publicFlags = Object.keys(Command.flags || {}).filter(flag => !Command.flags[flag].hidden).map(flag => `--${flag}`).join(' ')
+          const flags = publicFlags.length ? ` ${publicFlags}` : ''
+          const namespace = p.namespace ? `${p.namespace}:` : ''
+          const id = Command.command ? `${Command.topic}:${Command.command}` : Command.topic
+          return `${namespace}${id}${flags}`
+        })
       }))
       const commands = flatten(cmds).join('\n')
-      fs.writeFileSync(path.join(config.dataDir, 'client', 'node_modules', 'cli-engine', 'autocomplete', 'commands'), commands)
+      fs.writeFileSync(path.join((new this().completionsPath), 'commands'), commands)
```

Comment:
oh because it's static. Why is it static? This is a terrible idea, you lose the config object this way